src/azure-cli-core/azure/cli/core/tooling.py

```diff
@@ -0,0 +1,199 @@
+"""
+Tooling Integration
+
+This module is used by the Azure CLI Tools extension for VSCode. It should not be used
+anywhere else and will be removed in a future release.
+"""
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import print_function
+
+import os
+import traceback
+from importlib import import_module
+from sys import stderr
+import pkgutil
+import yaml
+
+from six.moves import configparser
+
+from azure.cli.core.application import APPLICATION, Configuration
+from azure.cli.core.commands import _update_command_definitions, BLACKLISTED_MODS
+from azure.cli.core._profile import _SUBSCRIPTION_NAME, Profile
+from azure.cli.core._session import ACCOUNT
+from azure.cli.core._environment import get_config_dir as cli_config_dir
+from azure.cli.core._config import az_config, GLOBAL_CONFIG_PATH, DEFAULTS_SECTION
+from azure.cli.core.help_files import helps
+from azure.cli.core.util import CLIError
+
+
+GLOBAL_ARGUMENTS = {
+    'verbose': {
+        'options': ['--verbose'],
+        'help': 'Increase logging verbosity. Use --debug for full debug logs.'
+    },
+    'debug': {
+        'options': ['--debug'],
+        'help': 'Increase logging verbosity to show all debug logs.'
+    },
+    'output': {
+        'options': ['--output', '-o'],
+        'help': 'Output format',
+        'choices': ['json', 'tsv', 'table', 'jsonc']
+    },
+    'help': {
+        'options': ['--help', '-h'],
+        'help': 'Get more information about a command'
+    },
+    'query': {
+        'options': ['--query'],
+        'help': 'JMESPath query string. See http://jmespath.org/ for more information and examples.'
+    }
+}
+
+
+def initialize():
+    _load_profile()
+
+
+def _load_profile():
+    azure_folder = cli_config_dir()
+    if not os.path.exists(azure_folder):
+        os.makedirs(azure_folder)
+
+    ACCOUNT.load(os.path.join(azure_folder, 'azureProfile.json'))
+
+
+def load_command_table():
+    APPLICATION.initialize(Configuration())
+    command_table = APPLICATION.configuration.get_command_table()
+    _install_modules(command_table)
+    return command_table
+
+
+def _install_modules(command_table):
```

Comment:
Once the extensions PR gets merged, this may have to change to handle loading of extensions as well. https://github.com/Azure/azure-cli/pull/4167/files#diff-34546b027e108d42bd46a0c08a1fe155R379